### PR TITLE
[rllib] Make timestep a required arg for exploration classes

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

By making the timestep arg required, we both (1) make the exploration class fully stateless in many cases, eliminating some (currently unimplemented) state management code, and (2) prevent mistakes where we forget to pass it in.

### CODE REVIEW

File: rllib/policy/eager_tf_policy.py [hunk 1]
```diff
@@ -332,8 +332,15 @@ def compute_actions(self,
...
-                    self.action_space, explore, self.config, timestep)
+                    self,
+                    self.model,
+                    input_dict,
+                    self.observation_space,
+                    self.action_space,
+                    explore,
+                    self.config,
+                    timestep=timestep
+                    if timestep is not None else self.global_timestep)
```
Comment (ericl) [inline rllib/policy/eager_tf_policy.py]: I think this was a bug where we never used the global timestep when eager was on.

File: rllib/utils/exploration/epsilon_greedy.py [hunk 2]
```diff
@@ -153,33 +148,5 @@ def _get_torch_exploration_action(self, q_values, explore, timestep):
...
-
-    @classmethod
-    @override(Exploration)
-    def merge_states(cls, exploration_objects):
-        timesteps = [e.get_state() for e in exploration_objects]
-        if exploration_objects[0].framework == "tf":
-            return tf.reduce_sum(timesteps)
-        return np.sum(timesteps)
+        eps = self.epsilon_schedule(self.last_timestep)
+        return {"cur_epsilon": eps}
```
Comment (ericl) [inline rllib/utils/exploration/epsilon_greedy.py]: I looked into trying to avoid making get_info() return TensorType in graph mode, but that seemed like a big change so punting on that.

File: rllib/utils/exploration/epsilon_greedy.py [hunk 3]
```diff
@@ -153,33 +148,5 @@ def _get_torch_exploration_action(self, q_values, explore, timestep):
...
     @override(Exploration)
     def get_info(self):
-        """Returns the current epsilon value.
-
-        Returns:
-            Union[float,tf.Tensor[float]]: The current epsilon value.
-        """
-        return self.epsilon_schedule(self.last_timestep)
-
-    @override(Exploration)
```
Comment (ericl) [inline rllib/utils/exploration/epsilon_greedy.py]: Removed this unimplemented state code for now. We can add it back as needed later.

File: rllib/utils/exploration/exploration.py [hunk 4]
```diff
@@ -66,21 +65,21 @@ def get_exploration_action(self,
         pass
 
     def get_loss_exploration_term(self,
-                                  model_output,
-                                  model=None,
-                                  action_dist=None,
-                                  action_sample=None):
+                                  model_output: TensorType,
```
Comment (ericl) [inline rllib/utils/exploration/exploration.py]: Which algo is using this loss bonus?

Comment (ericl) [inline rllib/utils/exploration/exploration.py]: Should we remove this unused method? Not sure if it's needed.

Comment (sven1977) [reply rllib/utils/exploration/exploration.py]: For now, yes. We won't need it for a while. Want to do param noise next.

**Review by sven1977 — APPROVED** [review]
Looks all good. Thanks!
